src/senaite/core/content/sampletemplate.py

```diff
@@ -0,0 +1,635 @@
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright 2018-2024 by it's authors.
+# Some rights reserved, see README and LICENSE.
+
+from AccessControl import ClassSecurityInfo
+from bika.lims import api
+from bika.lims import senaiteMessageFactory as _
+from bika.lims.interfaces import IDeactivable
+from plone.autoform import directives
+from plone.supermodel import model
+from Products.CMFCore import permissions
+from senaite.core.catalog import SETUP_CATALOG
+from senaite.core.config.widgets import get_default_columns
+from senaite.core.content.base import Container
+from senaite.core.content.mixins import ClientAwareMixin
+from senaite.core.interfaces import ISampleTemplate
+from senaite.core.schema import UIDReferenceField
+from senaite.core.schema.fields import DataGridRow
+from senaite.core.z3cform.widgets.datagrid import DataGridWidgetFactory
+from senaite.core.z3cform.widgets.listing.widget import ListingWidgetFactory
+from senaite.core.z3cform.widgets.uidreference import UIDReferenceWidgetFactory
+from zope import schema
+from zope.interface import Interface
+from zope.interface import implementer
+
+
+class IServiceRecord(Interface):
+    """Record schema for selected services and partitions
+    """
+    uid = schema.TextLine(title=u"Service UID")
+    hidden = schema.Bool(title=u"Hidden")
+    part_id = schema.TextLine(title=u"Partition ID")
+
+
+class IPartitionRecord(Interface):
+    """DataGrid Row for Sample Partition Schema
+    """
+
+    # PARTITION
+    directives.widget("part_id",
+                      style=u"width:100px!important")
+    part_id = schema.TextLine(
+        title=_(
+            u"label_sampletemplate_partition_part_id",
+            default=u"Partition ID"
+        ),
+        required=False,
+        default=u"part-1")
+
+    # CONTAINER
+    directives.widget(
+        "container",
+        UIDReferenceWidgetFactory,
+        catalog=SETUP_CATALOG,
+        query={
+            "portal_type": "SampleContainer",
+            "is_active": True,
+            "sort_on": "sortable_title",
+            "sort_order": "ascending",
+        },
+        display_template="<a href='${url}'>${Title}</a>",
+        columns=get_default_columns,
+        limit=5)
+    container = UIDReferenceField(
+        title=_(
+            u"label_sampletemplate_partition_container",
+            default=u"Container"
+        ),
+        allowed_types=("SampleContainer", ),
+        multi_valued=False,
+        required=False)
+
+    # PRESERVATION
+    directives.widget(
+        "preservation",
+        UIDReferenceWidgetFactory,
+        catalog=SETUP_CATALOG,
+        query={
+            "portal_type": "SamplePreservation",
+            "is_active": True,
+            "sort_on": "sortable_title",
+            "sort_order": "ascending",
+        },
+        display_template="<a href='${url}'>${Title}</a>",
+        columns=get_default_columns,
+        limit=5)
```

Comment:
I would remove `limit=5`, is already the limit set by default in `QuerySelectWidget`